packages/editor-ui/src/utils/__tests__/nodeTypesUtils.spec.ts

```diff
@@ -0,0 +1,283 @@
+				},
+				filtersUI: {},
+				combineFilters: 'AND',
+				options: {},
+			},
+			type: 'n8n-nodes-base.googleSheets',
+			typeVersion: 4.5,
+			position: [440, 0],
+			id: '9a95ad27-06cf-4076-af6b-52846a109a8b',
+			name: 'Google Sheets',
+			credentials: {
+				googleSheetsOAuth2Api: {
+					id: '8QEpi028oHDLXntS',
+					name: 'milorad@n8n.io',
+				},
+			},
+		},
+		expected: ['Edit Fields'],
+	},
+	{
+		caseName: 'Should handle multiple node references in one expression',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('Edit Fields').item.json.one }} {{ $('Edit Fields 2').item.json.two }} {{ $('Edit Fields').item.json.three }}",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: ['Edit Fields', 'Edit Fields 2'],
+	},
+	{
+		caseName: 'Should ignore spaces around node references',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('   Edit Fields     ').item.json.one }}",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: ['Edit Fields'],
+	},
+	{
+		caseName: 'Should ignore non-existing nodes',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('   Edit Fields     ').item.json.one }}",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: ['Edit Fields'],
+	},
+	{
+		caseName: 'Should ignore special characters in node references',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $(    'Ignore ' this'    ).item.json.document }",
+				authentication: 'none',
+				provideSslCertificates: false,
+				sendQuery: false,
+				sendHeaders: false,
+				sendBody: false,
+				options: {},
+				infoMessage: '',
+			},
+			type: 'n8n-nodes-base.httpRequest',
+			typeVersion: 4.2,
+			position: [220, 220],
+			id: 'edc36001-aee7-4052-b66e-cf127f4b6ea5',
+			name: 'HTTP Request',
+		},
+		expected: [],
+	},
+	{
+		caseName: 'Should handle node names with quotes',
+		node: {
+			parameters: {
+				curlImport: '',
+				method: 'GET',
+				url: "={{ $('Edit 'Fields' 2').item.json.name }}",
```

Comment:
this does not seem like a valid expression. Why would it be extracted? If you copy this into n8n, it should not work. has to be `$('Edit \'Fields\' 2')` to make sense